homeassistant/components/roborock/roborock_storage.py

```diff
@@ -0,0 +1,152 @@
+
+    def _save_map(self, filename: str, content: bytes) -> None:
+        """Help other functions save the map. Should not be called separately."""
+        os.makedirs(os.path.dirname(filename), exist_ok=True)
+        _LOGGER.debug("Saving event map to disk store: %s", filename)
+        with open(filename, "wb") as stored_map:
+            stored_map.write(content)
+
+    async def async_save_map(self, map_name: str, content: bytes) -> None:
+        """Write map if it should be updated."""
+        await self.async_save_maps([(map_name, content)])
+
+    async def async_save_maps(self, maps: list[tuple[str, bytes]]) -> None:
+        """Write maps - update regardless. Should be called as background task."""
+
+        def save_maps(maps: list[tuple[str, bytes]]) -> None:
+            for map_name, content in maps:
+                map_entry = self._data.get(map_name)
+                filename = self._get_map_filename(map_name)
+                self._data[map_name] = RoborockMapEntry(
+                    map_name, dt_util.utcnow().timestamp()
+                )
+                try:
+                    self._save_map(filename, content)
+                except OSError as err:
+                    _LOGGER.error("Unable to write map file: %s %s", filename, err)
+                    # We don't want the _data dict to be updated with incorrect information.
+                    if map_entry is not None:
+                        self._data[map_name] = map_entry
+
+        await self._hass.async_add_executor_job(save_maps, maps)
+
+    async def async_remove_maps(self) -> None:
+        """Remove all maps associated with a config entry."""
+
+        def remove_maps() -> None:
+            directory = self._hass.config.path(f"{MAP_PATH}/{self._entry_id}")
+            try:
+                for filename in os.listdir(directory):
+                    file_path = os.path.join(directory, filename)
+                    _LOGGER.debug("Removing map from disk store: %s", file_path)
+                    os.remove(file_path)
+            except OSError as err:
+                _LOGGER.error(
+                    "Unable to remove map files for: %s %s", self._entry_id, err
+                )
+
+        await self._hass.async_add_executor_job(remove_maps)
+
+    async def async_save_map_info(
+        self, coord_duid: str, map_info: dict[int, RoborockMapInfo]
+    ) -> None:
+        """Save the coordinator map info."""
+        self._map_info[coord_duid] = map_info
+        await self._store.async_save(self._map_info)
+
+    async def async_load_map_info(self, coord_duid: str) -> dict[int, RoborockMapInfo]:
+        """Load a coordinator's map info."""
+        map_info: dict[str, dict[str, dict]] | None = await self._store.async_load()
+        if map_info is None:
+            return {}
+        # Oddly enough the storage seems to turn the mapflag key into a str. This
+        # breaks logic downstream.
+        for stored_coord_duid in map_info:
```

Comment:
Iterate over `.items()` here also to get the value right away rather than doing the value lookup `map_info[stored_coord_duid]` below.